source/run-command/run-command.rst

```diff
@@ -187,8 +193,148 @@ Drivers MUST document the behavior of RunCommand if a ``maxTimeMS`` field  is al
 * See Client Side Operations Timeout's section on `runCommand <https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.rst#runcommand>`_
 * See Client Side Operations Timeout's section on `runCommand behavior <https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.rst#runcommand-behavior>`_
 
+
+--------------------
+``runCursorCommand``
+--------------------
+
+Drivers MAY expose a runCursorCommand API with the following syntax.
+
+.. code:: typescript
+
+    interface Database {
+      /**
+       * Takes an argument representing an arbitrary BSON document and executes it against the server.
+       */
+      runCursorCommand(command: BSONDocument, options: RunCursorCommandOptions): RunCommandCursor;
+    }
+
+    interface RunCursorCommandOptions extends RunCommandOptions {
+      /**
+       * For operations that create cursors, timeoutMS can either cap the lifetime of the cursor or be applied separately to the original operation and all subsequent calls.
+       * To support both of these use cases, these operations MUST support a timeoutMode option.
+       *
+       * @defaultValue 'cursorLifetime'
+       *
+       * @see https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.rst
+       */
+      timeoutMode?: 'iteration' | 'cursorLifetime';
+
+      /**
+       * Identifies the type of cursor this is.
+       *
+       * A tailable cursor can receive empty `nextBatch` arrays in `getMore` responses.
+       * However, subsequent `getMore` operations may return documents if new data has become available.
+       *
+       * A tailableAwait cursor is an enhancement where instead of dealing with empty responses the server will block until data becomes available.
+       *
+       * @defaultValue 'nonTailable'
+       */
+      cursorType: 'nonTailable' | 'tailable' | 'tailableAwait';
```

Comment:
Related to my comment on `timeoutMode`, it'd be helpful to explicitly state that this is an enum.

A CursorType enum is already defined in [CRUD: Read](https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#read) with values: `NON_TAILABLE`, `TAILABLE`, and `TAILABLE_AWAIT`. The Change Streams spec also refers `TAILABLE_AWAIT` enum value. I think it'd make sense to add a reference to the CursorType enum in the CRUD spec and use all-caps for the values here for consistency.

Regarding spec tests:

 * We don't have any spec tests for the `cursorType` `find` option, so there's no prior art for how to reference these enums in tests.
 * CSOT spec refer to `timeoutMode` enum values with `camelCase` despite using `ALL_CAPS` in the spec itself

Given that, I've no objection to leaving the `camelCase` in this PR's spec tests as-is.

Reply:
Updated same as CSOT, and I'll leave the tests as is